Packs/Axonius/Integrations/Axonius/axonius.py

```diff
@@ -0,0 +1,131 @@
+"""Demisto Integration for Axonius."""
+import traceback
+from typing import Any, List, Optional
+
+import axonius_api_client
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+# from CommonServerPython import (CommandResults, handle_proxy, return_error,
+#                                 return_results)
+
+
+MAX_ROWS: int = 50
+"""Maximum number of assets to allow user to fetch."""
+SKIPS: List[str] = ["specific_data.data.image"]
+"""Fields to remove from each asset if found."""
+
+
+def get_int_arg(
+    key: str, required: Optional[bool] = False, default: Optional[int] = None,
+) -> int:
+    """Get a key from a command arg and convert it into an int."""
+    args: dict = demisto.args()
+    value: int = args.get(key, default)
+
+    if value is None and required:
+        raise ValueError(f"No value supplied for argument {key!r}")
+
+    try:
+        return int(value)
+    except Exception:
+        raise ValueError(
+            f"Supplied value {value!r} for argument {key!r} is not an integer."
+        )
+
+
+def test_module(client: axonius_api_client.connect.Connect) -> str:
+    """Pass."""
+    client.start()
+    return "ok"
+
+
+def parse_kv(key: str, value: Any) -> Any:
+    """Pass."""
+    if "last_seen" in key:
+        try:
+            return axonius_api_client.tools.dt_parse(value).isoformat()
+        except Exception:
+            return value
+    return value
+
+
+def parse_key(key: str) -> str:
+    """Pass."""
+    if key.startswith("specific_data.data."):
+        key = axonius_api_client.tools.strip_left(obj=key, fix="specific_data.data.")
+        key = f"aggregated_{key}"
+    if key.startswith("adapters_data."):
+        key = axonius_api_client.tools.strip_left(obj=key, fix="adapters_data.")
+    key = key.replace(".", "_")
+    return key
+
+
+def parse_asset(asset: dict) -> dict:
+    """Pass."""
+    return {
+        parse_key(key=k): parse_kv(key=k, value=v)
+        for k, v in asset.items()
+        if k not in SKIPS
+    }
+
+
+def get_by_sq(
+    api_obj: axonius_api_client.api.assets.asset_mixin.AssetMixin,
+) -> CommandResults:
+    """Get assets with their defined fields returned by a saved query."""
+    args: dict = demisto.args()
+    name: str = args["saved_query_name"]
+    max_rows: int = get_int_arg(key="max_results", required=False, default=MAX_ROWS)
+
+    assets = api_obj.get_by_saved_query(name=name, max_rows=max_rows)
+    results = [parse_asset(asset=asset) for asset in assets]
+
+    return CommandResults(
+        outputs_prefix=f"Axonius.{api_obj.__class__.__name__}",
+        outputs_key_field="internal_axon_id",
+        outputs=results,
+    )
```

Comment:
You should provide a human readable outputs in `markdown` format, you should also make sure there's a human readable output when no assets are present. You can use `tableToMarkdown` from `CommonServerPython` script, See [tableToMarkdown Docs](https://xsoar.pan.dev/docs/integrations/code-conventions#tabletomarkdown) for more information.
Remember to differentiate users hr from devices hr.
```suggestion
        readable_output=hr
    )
```

Reply:
@guyfreund @fvigo said that our data would be automatically formatted. We went through a few examples together to make sure it formatted correctly. During that call, we collectively decided not to manually setup the readable output.